api/analyzers/javascript/analyzer.py

```
@@ -0,0 +1,110 @@
+            return None
+        raise ValueError(f"Unknown entity type: {node.type}")
+
+    def get_entity_types(self) -> list[str]:
+        return ['function_declaration', 'class_declaration', 'method_definition']
+
+    def add_symbols(self, entity: Entity) -> None:
+        if entity.node.type == 'class_declaration':
+            heritage = entity.node.child_by_field_name('body')
+            if heritage is None:
+                return
+            superclass_node = entity.node.child_by_field_name('name')
```

Ruff (0.15.1)

[error] Local variable `superclass_node` is assigned to but never used

Remove assignment to unused variable `superclass_node`

(F841)